tests/subresource_api_test.go

```diff
@@ -237,6 +241,164 @@ var _ = Describe("[sig-compute]Subresource Api", func() {
 				}, 90*time.Second, 1*time.Second).Should(Equal(v1.Running))
 			})
 		})
+
+		Context("ExpandInstancetype endpoint", func() {
+			Context("instancetype", func() {
+				var (
+					instancetype *instancetypev1alpha1.VirtualMachineInstancetype
+					expectedCpu  *v1.CPU
+					vmi          *v1.VirtualMachineInstance
+				)
+
+				BeforeEach(func() {
+					instancetype = newVirtualMachineInstancetype(nil)
+					instancetype.Spec.CPU.Guest = 2
+
+					expectedCpu = &v1.CPU{
+						Sockets: 2,
+						Cores:   1,
+						Threads: 1,
+					}
+
+					var err error
+					instancetype, err = virtCli.VirtualMachineInstancetype(util.NamespaceTestDefault).
+						Create(context.Background(), instancetype, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					vmi = libvmi.NewCirros()
+					// Removing memory requests, because they are set in instancetype
+					vmi.Spec.Domain.Resources.Requests = nil
+				})
+
+				AfterEach(func() {
+					err := virtCli.VirtualMachineInstancetype(util.NamespaceTestDefault).
+						Delete(context.Background(), instancetype.Name, metav1.DeleteOptions{})
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("[test_id:TODO] should return unchanged VirtualMachine, if instancetype is not used", func() {
+					vm := tests.NewRandomVirtualMachine(vmi, false)
+
+					vm.Spec.Template.Spec.Domain.CPU = &v1.CPU{Sockets: 2, Cores: 1, Threads: 1}
+
+					vm, err := virtCli.VirtualMachine(util.NamespaceTestDefault).Create(vm)
+					Expect(err).ToNot(HaveOccurred())
+
+					expandedVm, err := virtCli.VirtualMachine(util.NamespaceTestDefault).
+						GetWithInstancetype(vm.GetName())
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(expandedVm.Spec).To(Equal(vm.Spec))
+				})
+
+				It("[test_id:TODO] should return VirtualMachine with instancetype expanded", func() {
+
+					vm := tests.NewRandomVirtualMachine(vmi, false)
+					vm.Spec.Instancetype = &v1.InstancetypeMatcher{
+						Name: instancetype.Name,
+						Kind: "VirtualMachineInstancetype",
```

Comment:
supernit - can you use the constants from `staging/src/kubevirt.io/api/instancetype/register.go` ?

Reply:
done